src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -298,12 +300,18 @@ private bool TryNavigateToMappedFile(Workspace workspace, Document generatedDocu
             return false;
         }
 
-        private static MappedSpanResult? GetMappedSpan(ISpanMappingService spanMappingService, Document generatedDocument, TextSpan textSpan)
+        private MappedSpanResult? GetMappedSpan(ISpanMappingService spanMappingService, Document generatedDocument, TextSpan textSpan)
         {
-            var results = System.Threading.Tasks.Task.Run(async () =>
+            // Mappings for opened razor files are retrieved via the LSP client making a request to the razor server.
+            // If we wait for the result on the UI thread, we will hit a bug in the LSP client that brings us to a code path
+            // using ConfigureAwait(true).  This deadlocks as it then attempts to return to the UI thread which is already blocked by us.
+            // Instead, we invoke this in JTF run which should run the request on the threadpool thread and will mitigate deadlocks
+            // if anyone else tries to switch to the UI thread.
+            // Link to LSP client bug for ConfigureAwait(true) - https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1216657
+            var results = _threadingContext.JoinableTaskFactory.Run(async () =>
```

Comment:
```suggestion
            var results = _threadingContext.JoinableTaskFactory.Run(() =>
```

And also remove the awaiting; it's not doing anything here.